### PR TITLE
Fix for developer mode issue.

### PR DESCRIPTION
**Pull request by ivan-gomes**

### CODE REVIEW

**Comment by Enquier** [comment]
Previously the logic deleted any responseFile deep inside the` MMSUtils.sendMMSRequest` method. However in most cases this responseFile bubbled up to the usage layer and parsed via Jackson. Now instead of deleting this file immediately it's deletion is scheduled in the "close()" method of the JsonParser. Since JsonParser is AutoClosable, and all usages of JsonParser are resources on a try block the parser method (i.e our file cleanup step) should execute at the end of the try/catch block.

File: gradle.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version=5.0.0
+version=5.0.1
```
Comment (ivan-gomes) [inline gradle.properties:1]: Revert.

File: src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParser.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package gov.nasa.jpl.mbee.mdk.json.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.util.JsonParserDelegate;
+import gov.nasa.jpl.mbee.mdk.options.MDKOptionsGroup;
+
+import java.io.File;
+import java.io.IOException;
+
+public class MMSJsonParser extends JsonParserDelegate {
```
Comment (ivan-gomes) [inline src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParser.java:10]: Let's use a more descriptive name, e.g. `TempFileJsonParser`.

File: src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package gov.nasa.jpl.mbee.mdk.json.util;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.io.IOContext;
+import com.fasterxml.jackson.databind.MappingJsonFactory;
+
+import java.io.*;
+
+public class MMSJsonParserFactory extends MappingJsonFactory {
```
Comment (ivan-gomes) [inline src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java:10]: Let's use a more descriptive name, e.g. `TempFileParserFactory`.

File: src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public JsonParser createParser(File f) throws IOException, JsonParseException {
+        // true, since we create InputStream from File
+        IOContext ctxt = _createContext(f, true);
+        InputStream in = new FileInputStream(f);
+        MMSJsonParser parser = (MMSJsonParser) _createParser(_decorate(in, ctxt), ctxt);
```
Comment (ivan-gomes) [inline src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java:17]: Let's find an alternative that doesn't require a blind cast.

File: src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        parser.configureClose(f);
+        return parser;
+    }
+
+    @Override
+    protected JsonParser _createParser(InputStream in, IOContext ctxt) throws IOException {
```
Comment (ivan-gomes) [inline src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java:23]: All of the non-(temp)file usages are out of scope for this custom factory and should be left unimplemented, so future development uses the appropriate - default - factory when appropriate by failing early when this one is used.

**Comment by ivan-gomes** [comment]
@Enquier can you please re-test with the latest revisions?

**Review by Enquier — APPROVED** [review]
Looks good and tested to work as expected

**Pull request merged**